c/progress/progress_manager.h

```diff
@@ -53,6 +56,20 @@ class progress_manager {
     // can result in a segfault.
     mutable std::mutex mutex;
 
+    // This flag is initially set to `false`. When SIGINT signal is caught,
+    // `caught_interrupt` is set to `true`. When progress manager handles
+    // the interrupt, it also resets `caught_interrupt` back to `false`.
+    mutable bool caught_interrupt;
+
+    // This flag is kind of similar to the above, but is atomic and not
+    // cleared by the interrupt handler. It is only cleared when threads are
+    // joining, and used in the cases when there is no special scheduler
+    // (like in the cases of `parallel_for_static`, `nested_for_static`
+    // or a single threaded `parallel_for_ordered`) but we still want
+    // to abort execution of all threads when SIGINT is received.
+    mutable std::atomic<bool> abort_execution;
```

Comment:
I'm still not sure I understand the purpose of this flag fully.
I see that it is set together with `caught_interrupt`, but cleared at a later stage. But then, why do we need to clear `caught_interrupt` so early? Would it work if we removed `caught_interrupt` altogether, and left just this flag?
What about other, non-interrupt exceptions?

Reply:
Replaced these two flags with just one, hopefully its purpose is clear now.